### PR TITLE
Let products be rendered directly in AMP themes.

### PR DESCRIPTION
**Pull request by arthurlorenz**

### CODE REVIEW

File: src/ProductService.php [hunk 1]
```diff
@@ -317,4 +332,66 @@ protected function increaseTodaysRequestCount($increment = 1) {
     return $count['count'];
   }
 
+  /**
```
Comment (mbm80) [inline src/ProductService.php:341]: missing description

File: src/ProductServiceTrait.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * @file
+ *
+ */
+
+namespace Drupal\amazon_product_widget;
+
+trait ProductServiceTrait {
+  /**
```
Comment (mbm80) [inline src/ProductServiceTrait.php:10]: empty line missing

File: src/ProductServiceTrait.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * @file
+ *
+ */
```
Comment (mbm80) [inline src/ProductServiceTrait.php]: not needed

File: src/ProductServiceTrait.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Drupal\amazon_product_widget;
+
+trait ProductServiceTrait {
```
Comment (mbm80) [inline src/ProductServiceTrait.php:8]: missing doc

File: src/ProductService.php [hunk 5]
```diff
@@ -317,4 +338,71 @@ protected function increaseTodaysRequestCount($increment = 1) {
...
+   * @param string $entity_type
+   *   The entity type.
+   * @param string $entity_id
+   *   The entity id.
+   * @param string $fieldname
+   *   The field name.
+   *
+   * @return mixed[]
+   *   Build array.
+   * @throws \Drupal\amazon_product_widget\Exception\AmazonRequestLimitReachedException
```
Comment (mbm80) [inline src/ProductService.php:354]: missing empty line above

File: src/ProductService.php [hunk 6]
```diff
@@ -84,13 +101,17 @@ class ProductService {
...
    */
-  public function __construct(ProductStoreFactory $store_factory, StateInterface $state, LockBackendInterface $lock, ConfigFactoryInterface $config, QueueInterface $queue) {
+  public function __construct(ProductStoreFactory $store_factory, StateInterface $state, LockBackendInterface $lock, ConfigFactoryInterface $config, QueueInterface $queue, EntityTypeManager $entityTypeManager) {
     $this->productStore = $store_factory->get(ProductStore::COLLECTION_PRODUCTS);
     $this->state = $state;
     $this->lock = $lock;
     $this->queue = $queue;
+    $this->entityTypeManager = $entityTypeManager;
 
+    $this->settings = $config->get('amazon_product_widget.settings');
```
Comment (mbm80) [inline src/ProductService.php:114]: is this setting used anywhere?

File: src/ProductService.php [hunk 7]
```diff
@@ -15,6 +18,20 @@
...
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Amazon product widget settings.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $settings;
```
Comment (mbm80) [inline src/ProductService.php:33]: not sure if needed

File: src/Plugin/Field/FieldFormatter/AmazonProductFieldFormatter.php [hunk 8]
```diff
@@ -44,8 +48,42 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
       '#field' => $field->getParent()->getName(),
     ];
 
+    if ($this->getSetting('render_inline')) {
+      /** @var \Drupal\amazon_product_widget\ProductService $product_service */
```
Comment (mbm80) [inline src/Plugin/Field/FieldFormatter/AmazonProductFieldFormatter.php]: wrong var declaration, not needed anyways

File: src/ProductService.php (hunk 6 above)
Comment (arthurlorenz) [reply src/ProductService.php:114]: yes, line 390

**Pull request merged**